src/sentry/profiles/task.py

```diff
@@ -321,38 +332,45 @@ def _normalize_profile(profile: Profile, organization: Organization, project: Pr
 @metrics.wraps("process_profile.normalize")
 def _normalize(profile: Profile, organization: Organization) -> None:
     profile["retention_days"] = quotas.get_event_retention(organization=organization) or 90
+    platform = profile["platform"]
+    version = profile.get("version")
+
+    if platform not in {"cocoa", "android"} or version == "2":
+        return
 
-    if profile["platform"] in {"cocoa", "android"}:
-        classification_options = dict()
+    classification_options = dict()
 
-        if profile["platform"] == "android":
-            classification_options.update(
-                {
-                    "cpu_frequencies": profile["device_cpu_frequencies"],
-                    "physical_memory_bytes": profile["device_physical_memory_bytes"],
-                }
-            )
+    if platform == "android":
+        classification_options.update(
+            {
+                "cpu_frequencies": profile["device_cpu_frequencies"],
+                "physical_memory_bytes": profile["device_physical_memory_bytes"],
+            }
+        )
 
-        if "version" in profile:
-            device_options = {
+    if version == "1":
+        classification_options.update(
+            {
                 "model": profile["device"]["model"],
                 "os_name": profile["os"]["name"],
                 "is_emulator": profile["device"]["is_emulator"],
             }
-        else:
-            device_options = {
+        )
+    elif version is None:
+        classification_options.update(
+            {
                 "model": profile["device_model"],
                 "os_name": profile["device_os_name"],
                 "is_emulator": profile["device_is_emulator"],
             }
+        )
 
-        classification_options.update(device_options)
-        classification = str(classify_device(**classification_options))
+    classification = str(classify_device(**classification_options))
 
-        if "version" in profile:
-            profile["device"]["classification"] = classification
-        else:
-            profile["device_classification"] = classification
+    if version == "1":
```

Comment:
In some places above, we check for `!= "2"`. Should all checks be `== "1"` to be safe & consistent?

Reply:
Not necessarily. `!= "2"` covers Sample and Legacy formats profiles, `== "1"` would just cover the Sample format.